runtime/autoload/health/provider.vim

```diff
@@ -148,7 +141,7 @@ endfunction
 function! s:check_python(version) abort
   call health#report_start('Python ' . a:version . ' provider')
 
-  let python_bin_name = 'python'.(a:version == 2 ? '2' : '3')
+  let python_bin_name = 'python'.(a:version == 2 ? '' : '3')
```

Comment:
@phanimahesh The logic is kind of sloppy, `python_bin_name` gets overwritten after line 149. So I think this is fine.